modules/openy_gc_auth/modules/openy_gc_auth_personify/src/Plugin/GCIdentityProvider/Personify.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+namespace Drupal\openy_gc_auth_personify\Plugin\GCIdentityProvider;
+
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Link;
+use Drupal\Core\Url;
+use Drupal\openy_gc_auth\GCIdentityProviderPluginBase;
+
+/**
+ * Personify SSO identity provider plugin.
+ *
+ * @GCIdentityProvider(
+ *   id="personify",
+ *   label = @Translation("Personify provider"),
+ *   config="openy_gc_auth.provider.personify"
+ * )
+ */
+class Personify extends GCIdentityProviderPluginBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function defaultConfiguration():array {
+    return [
+      'api_login_check' => '/openy_gc_auth_personify/check/login',
+      'api_get_login_url' => '/openy_gc_auth_personify/get-login-url',
+      'api_auth' => '/openy_gc_auth_personify/auth',
+      'api_logout' => '/openy_gc_auth_personify/logout',
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildConfigurationForm(array $form, FormStateInterface $form_state) {
+    $config = $this->getConfiguration();
+    $form = parent::buildConfigurationForm($form, $form_state);
+
+    $personifyConfigForm = Link::createFromRoute($this->t('Personify configuration form'),
+      'personify.settings', [], ['target' => '_blank'])->toString();
+    $readMeUrl = Link::fromTextAndUrl($this->t('README file'),
+      Url::fromUserInput('/' . drupal_get_path('module', 'openy_gc_auth_personify') . '/README.md',
+        ['attributes' => ['target' => '_blank']])
+    )->toString();
+
+    $form['help'] = [
+      '#type' => 'container',
+      'text' => [
+        '#theme' => 'item_list',
+        '#items' => [
+          $this->t('Setup Personify credentials in settings.php based on @readme.', ['@readme' => $readMeUrl]),
+          $this->t('Select environment on @form.', ['@form' => $personifyConfigForm]),
+        ],
+      ],
+    ];
+
+    $form['endpoints'] = [
+      '#type' => 'details',
+      '#title' => t('Custom Personify endpoints'),
+      '#collapsible' => TRUE,
+      '#open' => FALSE,
+      '#description' => $this->t('Change these values only in case you have custom endpoints for every URL.'),
+    ];
+    $form['endpoints']['api_login_check'] = [
+      '#title' => $this->t('Login check endpoint'),
```

Comment:
There is no need to set routes that you've created yourself.
These routes were created here: https://github.com/fivejars/openy_gated_content/pull/31/files#diff-6bbd677965587a158ae13f2d460bd247 and there is no need to ask administrator to enter them again.
All that we need here is to save Virtual Ymca page, where you will redirect user at the end.